### PR TITLE
Issue 39

### PR DESCRIPTION
**Pull request by phillbarber**
Adds support for /tests endpoint.  See https://github.com/atam4j/atam4j/issues/39

### CODE REVIEW

**Comment by anuragkapur** [comment]
Instead of having 2 separate config files (`app-config-with-one-test.yml` and `app-config-with-passing-and-failing-test.yml`)for the sample app used in acceptance tests, can we not avoid config duplication and pass the testClasses as a `ConfigOverride`?

**Comment by phillbarber** [comment]
I would love to. That would be way simpler. However, when you use the
configuration override i couldn't find a way to add an additional item to
an array.

If you can crack that then please go ahead!

Pergola Software Limited

Company number:  09275615 (incorporated in England and Wales)

Registered office:  5 Beaumont Gate, Shenley Hill,  Radlett,  WD7 7AR

**Comment by anuragkapur** [comment]
@phillbarber - yeah, realised ConfigOverride for adding new elements to array was tricky / not-doable. Suggested an alternate approach in this PR: https://github.com/atam4j/atam4j/pull/60.

Not sure if this is any better, so keen to hear what you think. If it is even slightly better, we can use it.

File: atam4j/src/main/java/me/atam/atam4j/TestRunListener.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    @Override
+    public void testStarted(Description description) throws Exception {
+        individualTestReportMap.put(new TestIdentifier(description), new IndividualTestResult(description.getClassName(), description.getMethodName(), true));
+    }
+
+    @Override
+    public void testAssumptionFailure(Failure failure) {
+        //TODO what does this mean?
```
Comment (anuragkapur) [inline atam4j/src/main/java/me/atam/atam4j/TestRunListener.java]: Pending todo?

File: atam4j/src/main/java/me/atam/atam4j/TestRunListener.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
+package me.atam.atam4j;
+
+import me.atam.atam4jdomain.IndividualTestResult;
+import me.atam.atam4jdomain.TestsRunResult;
+import org.junit.runner.Description;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunListener;
+
+import java.util.*;
```
Comment (anuragkapur) [inline atam4j/src/main/java/me/atam/atam4j/TestRunListener.java]: IMO, wildcard imports should be avoided.

File: atam4j/src/main/java/me/atam/atam4j/TestRunListener.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    @Override
+    public void testAssumptionFailure(Failure failure) {
+        //TODO what does this mean?
+    }
+
+    @Override
+    public void testFailure(Failure failure) throws Exception {
+        IndividualTestResult individualTestResult = new IndividualTestResult(failure.getDescription().getClassName(), failure.getDescription().getMethodName(), false);
+        TestIdentifier k = new TestIdentifier(failure.getDescription());
+        IndividualTestResult put = individualTestReportMap.put(k, individualTestResult);
```
Comment (anuragkapur) [inline atam4j/src/main/java/me/atam/atam4j/TestRunListener.java]: variable `put` is unused and can be removed...

**Comment by anuragkapur** [comment]
Looks good. :+1: 

Have made a few minor changes (formatting and cleanup like removal of unused var, import) and submitted a PR to this PR to get your opinion on https://github.com/atam4j/atam4j/pull/59#issuecomment-161329552

**Pull request merged**